src/tests/test_x_vesync_whogplug.py

```
@@ -0,0 +1,112 @@
+        assert whogplug_outlet.uuid == call_json.Defaults.uuid(DEVICE_TYPE)
+
+    def test_whogplug_details(self):
+        """Test WhogPlug get_details()."""
+        self.mock_api.return_value = CORRECT_WHOGPLUG_DETAILS
+        whogplug_outlet = VeSyncOutletWhogPlug(DEV_LIST_DETAIL, self.manager)
+        whogplug_outlet.get_details()
+        response = CORRECT_WHOGPLUG_DETAILS[0]
+        result = response.get('result', {})
+        
+        expected_status = 'on' if result.get('enabled') else 'off'
+        assert whogplug_outlet.device_status == expected_status
```

Ruff (0.12.2)

Use of `assert` detected

(S101)